### PR TITLE
Server Profile OS Deployment

### LINKED ISSUE #217: Create full server deploy example recipe using Image Streamer and OneView resources
### Scenario/Intent
I want to have an example on how to perform the deployment of a server using the new Image Streamer resources.

### Environment Details
 - **oneview Cookbook Version:** 2.2.0
 - **OneView Version:** API300

### Steps to Reproduce
Create one example recipe to show how to deploy/provision a server.

### Expected Result
I would have an example recipe inside `examples/` folder.

### Actual Result
There's no clue on how to do it.

### PR DESCRIPTION
**Pull request by tmiotto**
### Description
Add support to OS Deployment Plans in the oneview_server_profile resource.

### Issues Resolved
#217

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: examples/image_streamer/server_profile_deploy.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+# Create or update the Deployment Plan with the OS build plan and the golden image
+image_streamer_deployment_plan 'DeploymentPlan1' do
+  client i3s_client
+  data(hpProvided: false)
+  os_build_plan 'ESXi - Build simple environment'
+  golden_image 'GoldenImage - Deploy1'
+end
+
+# Instantiate a new server profile from the previously created template 'WebServerTemplate1'
```
Comment (fgbulsoni) [inline examples/image_streamer/server_profile_deploy.rb]: Would probably be good to add a comment specifying that this SPT must be created prior to running the cookbook, maybe somewhere along the start of the cookbook?

**Review by fgbulsoni — APPROVED** [review]
Overall looks good. I'd suggest adding a comment at the start of the example file about what is required for this example to be run, but other than that looking good :sparkles: :sparkles:

File: examples/image_streamer/server_profile_deploy.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  profile_name 'ESXi - WebServer1'
+  action :new_profile
+end
+
+# Apply the server profile to the 'SY0000, bay 1' server hardware blade with the OS deployment plan
+oneview_server_profile 'ESXi - WebServer1' do
+  client oneview_client
+  os_deployment_plan 'ESXi - Build simple environment'
+  server_hardware 'SY0000, bay 1'
+end
```
Comment (jsmartt) [inline examples/image_streamer/server_profile_deploy.rb]: Can you do these last 2 resources with just the oneview_server_profile resource? Something like:

```ruby
oneview_server_profile 'ESXi - WebServer1' do
  client oneview_client
  server_profile_template 'WebServerTemplate1'
  server_hardware 'Enclosure1, bay 1'
  os_deployment_plan 'ESXi - Build simple environment'
end
```

File: README.md [hunk 3]
```diff
@@ -745,13 +745,15 @@ oneview_server_profile 'ServerProfile1' do
...
   network_set_connections <network_set_connections_data>
+  deployment_plan <image_streamer_deployment_plan_name>
   action [:create, :create_if_missing, :delete]
 end
 ```
 
 You can specify the association of the server profile with each of the resources using the resource properties. Also it is easy to add connections using the connection properties:
 
 - **\<network_type\>_connections** (Hash) Optional - Specify connections with the desired resource type. The Hash should have `<network_name> => <connection_data>` associations. See the [examples](examples/server_profile.rb) for more information.
+- **deployment_plan** (String) Optional - Specify the Deployment Plan to be applied with the Server Profile. The OS Deployment Plan need to be created in Image Streamer appliance. See the [examples](examples/image_streamer/server_profile_deploy.rb) for more information.
```
Comment (jsmartt) [inline README.md]: Make a note about this property being supported only in API300+. Also, change the link text from "examples" to "example"

**Review by jsmartt — CHANGES_REQUESTED** [review]

**Review by jsmartt — APPROVED** [review]
👍

**Pull request merged**